psamm_import/importer.py

```diff
@@ -168,19 +171,95 @@ def model_compounds(model):
         yield d
 
 
-def model_reactions(model, exchange=False):
-    """Yield model reactions as YAML dicts"""
+def reactions_to_files(model, dest, yaml_args, exchange,
+                       single_reaction_file):
+    """Turns the reaction subsystems into their own files.
 
-    for reaction_id, reaction in sorted(iteritems(model.reactions)):
+    If a subsystem has a number of reactions over the threshold, it gets its
+    own yaml file. All other reactions, those that don't have a subsystem or
+    are in a subsystem that falls below the threshold, get added to a common
+    reaction file.
+
+    Args:
+        model: MetabolicModel from psamm-importer.
+        dest: output path for model files.
+        yaml_args: yaml style format arguments
+        exchange: flag for whether to include exchange reactions
+    """
+
+    common_reactions = []
+    reaction_files = []
+    if single_reaction_file:
+        for reaction_id, reaction in sorted(iteritems(model.reactions)):
+            common_reactions.append(reaction)
+        reactions_dump = list(
+            model_reactions(common_reactions, exchange=exchange))
+        if len(reactions_dump) > 0:
+            reaction_file = 'reactions.yaml'
+            with open(os.path.join(dest, reaction_file), 'w') as f:
+                yaml.safe_dump(reactions_dump, f, **yaml_args)
+            reaction_files.append(reaction_file)
+    else:
+        subsystems = {}
+        for reaction_id, reaction in sorted(iteritems(model.reactions)):
+            if reaction.subsystem is not None:
+                subsystem_file = re.sub(
+                    r'\W+', '_', reaction.subsystem, flags=re.UNICODE)
+                subsystem_file = re.sub(
+                    r'_+', '_', subsystem_file.lower(), flags=re.UNICODE)
```

Comment:
I would extract these `re.sub(...` lines to a small function. Maybe call it `safe_file_name`.